Add docs pages (#167)
saleor/react-storefront: docs/translations.md

# Translations

> There are 2 types of translations:

Reviewer: ```suggestion
There are two types of translations:
```

> - translations which can be managed via Dashboard and are fetched from the API. For example: name of a certain product

Reviewer: ```suggestion
- Translations that can be managed via Dashboard and are fetched from the API. For example, the name of a specific product.
```

> - interface translations. For example label on "Add to cart" button

Reviewer: ```suggestion
- Interface translations. For example, the label on the "Add to cart" button.
```

## API translations

> Documentation for API can be found [here](https://docs.saleor.io/docs/3.x/developer/api-conventions/translations). Dashboard guide is located [here](https://docs.saleor.io/docs/3.x/dashboard/translations).

Reviewer: ```suggestion
Documentation for the API can be found [here](https://docs.saleor.io/docs/3.x/developer/api-conventions/translations). The dashboard guide is located [here](https://docs.saleor.io/docs/3.x/dashboard/translations).
```

> React Storefront has helper for displaying translated field called `translate` located in th `lib/translations.ts` file. Function will fall back to default language, if API has no translation available.

Reviewer: ```suggestion
React Storefront has a helper for displaying translated fields called `translate` located in the `lib/translations.ts` file. The function will revert back to the default language if the API has no translation available.
```

## Interface translations

> SRS use [FormatJS](https://formatjs.io/) as a library for translating the interface. If you haven't used it before, we recommend reading it's [documentation](https://formatjs.io/docs/getting-started/application-workflow).

Reviewer: ```suggestion
SRS uses [FormatJS](https://formatjs.io/) as a library for translating the interface. We recommend reading its [documentation](https://formatjs.io/docs/getting-started/application-workflow) if you haven't used it before.
```

> Translation sources are located at `locale` directory. Please note: SRS use full language code names to differentiate between language dialects (eg. `en-US` and `en-UK`).

Reviewer: ```suggestion
Translation sources are located in the `locale` directory. Please note that SRS uses full language code names to differentiate between language dialects (e.g., en-US and `en-UK`).
```

### Adding new languages

1. Create a new json file in the `locale` directory
2. Add a language to the `LOCALES` constant in the `lib/regions.ts` file
3. Extend the `importMessages` function to import the proper json file based on chosen locale

### Translation management

> Instead of editing file by hand, we recommend using translation platforms like:

Reviewer: ```suggestion
Instead of editing files manually, we recommend using translation platforms like:
```
